src/TextInputWidget.cpp

```diff
@@ -15,22 +15,29 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <QAbstractItemView>
 #include <QAbstractTextDocumentLayout>
 #include <QBuffer>
 #include <QClipboard>
+#include <QCompleter>
 #include <QFileDialog>
 #include <QMimeData>
 #include <QMimeDatabase>
 #include <QMimeType>
 #include <QPainter>
 #include <QStyleOption>
 #include <QtConcurrent>
+#include <qnamespace.h>
+#include <qregexp.h>
```

Comment:
Can you clean up the includes a bit. I think you only need to add QCompleter here and QRegularExpression. The .h version are something like internal includes afaik.